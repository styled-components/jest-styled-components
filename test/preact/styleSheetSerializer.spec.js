import { mount, shallow } from 'enzyme';
import React from 'react';
import styled from 'styled-components';

const toMatchSnapshot = component => {
  expect(shallow(component)).toMatchSnapshot('shallow');
  expect(mount(component)).toMatchSnapshot('mount');
};

it('basic', () => {
  const Wrapper = styled.section`
    padding: 4em;
    background: papayawhip;
  `;

  const Title = styled.h1`
    font-size: 1.5em;
    text-align: center;
    color: palevioletred;
  `;

  toMatchSnapshot(
    <Wrapper>
      <Title>Hello World, this is my first styled component!</Title>
    </Wrapper>
  );
});
