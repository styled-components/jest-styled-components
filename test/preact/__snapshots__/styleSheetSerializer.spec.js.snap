// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic: mount 1`] = `
.c0 {
  padding: 4em;
  background: papayawhip;
}

<styled.section>
  <section
    className="c0"
  >
    <styled.h1>
      Hello World, this is my first styled component!
    </styled.h1>
  </section>
</styled.section>
`;

exports[`basic: shallow 1`] = `
.c0 {
  padding: 4em;
  background: papayawhip;
}

<section
  className="c0"
>
  <styled.h1>
    Hello World, this is my first styled component!
  </styled.h1>
</section>
`;
